CUSTOM_IMAGE.md

```diff
@@ -0,0 +1,76 @@
+Building a Custom Spark Image
+=============================
+
+Overview
+--------
+To build a custom Spark image, with a custom version of Spark or other libraries, 
+such as Hadoop, Python, or R, follow the steps below. This build does the following:
+1. Builds a Docker image containing Spark and other libraries.
+1. Builds a DC/OS package (also called a "universe") based on this Docker image.
+1. Uploads this package to Amazon S3.
+
+Prerequisites
+-------------
+* [AWS Command Line Interface](https://aws.amazon.com/cli/) (CLI)
+* The [jq](https://github.com/stedolan/jq) JSON parser
+
+Custom Spark Version, Hadoop Version
+------------------------------------
+Edit `manifest.json`, and set `spark_uri` to point to a custom Spark build. This custom build may 
+contain a custom version of Spark or Hadoop.
+To build a custom version of Spark, run
+`./dev/make-distribution.sh` in your Spark project directory, according to the instructions
+[here](http://spark.apache.org/docs/latest/building-spark.html#building-a-runnable-distribution). 
+Make sure to set the sparkr flag if you will be using SparkR.
+
+Custom Python, R Versions
+-------------------------
+Edit `docker/Dockerfile` to install custom versions of Python or R in the Docker image.
+
+To install a custom version of Python: the base Docker image already comes with a 
+version of Python. [Add instructions to replace with another Python version ...]
+
+To install a custom version of R, replace the lines 
+```
+RUN apt-get update && \
+    apt-get install -y r-base
+```
+with the instructions to install a custom version of R.
+
+Run the Custom Build
+--------------------
+```
+DOCKER_IMAGE=<docker-name> S3_BUCKET=<bucket> S3_DIR_PATH=<dir> make universe 
+```
+
+At the end of the build output, you will see the URL to the newly built stub universe.
+
```

Comment:
"You will see the URL to the newly built stub universe at the end of the build output"